src/envs/behavior.py

```diff
@@ -166,19 +165,30 @@ def _generate_train_tasks(self) -> List[Task]:
         return self._get_tasks(num=CFG.num_train_tasks, rng=self._train_rng)
 
     def _generate_test_tasks(self) -> List[Task]:
-        return self._get_tasks(num=CFG.num_test_tasks, rng=self._test_rng)
-
-    def _get_tasks(self, num: int, rng: np.random.Generator) -> List[Task]:
+        return self._get_tasks(num=CFG.num_test_tasks,
+                               rng=self._test_rng,
+                               testing=True)
+
+    def _get_tasks(self,
+                   num: int,
+                   rng: np.random.Generator,
+                   testing: bool = False) -> List[Task]:
         tasks = []
+        assert num <= 10  # Max 10 train and test tasks for behavior
+        if testing:
+            self.task_num += 10
```

Comment:
What's confusing me about task num is that it's incremented all at once here. If you're using it to identify different tasks, why does this work? I would have thought that you would increment it one by one after each task is created.

Reply:
Good catch I think it should be `self.task_num = 10`. So if `behavior_randomize_init_state = True` it will also need to act as the ID of the current environment instance which will go [0...N] for N train tasks and [10...10+M] for M train tasks

Reply:
Two more questions based on that:
1. Can we set it to a CFG value instead of just 10? Like can it just be CFG.num_train_tasks instead?
2. Is there a way to make it not a class variable and to just have it be a variable in this function? Or where else is it gonna be importantly used?

Reply:
Yes we can do 1 where `self.task_num = CFG.max_num_train_tasks`. We use it when we load in different scenes. The task_num changes based on what scene is loaded.